src/Marten/Services/SessionOptions.cs

```diff
@@ -93,6 +101,23 @@ internal IConnectionLifetime Initialize(DocumentStore store, CommandRunnerMode m
             throw new DefaultTenantUsageDisabledException();
         }
 
+        var innerConnectionLifetime =GetInnerConnectionLifetime(store, mode);
+
+        var corelationId = Activity.Current is not null ? Activity.Current.RootId : Guid.NewGuid().ToString();
```

Comment:
When there's no activity that is recored (`Current = null`), is it OK to just create a new id?
I don't know what's best here.

Reply:
As far as I understand things, if the current activity is null, this means you are the first activity in the chain, so creating a new activity/id is the right thing to do.

Reply:
> if the current activity is null, this means you are the first activity in the chain

Being the first one in the chain means that there's no parent, but `Current` is set an `Activity`.

When there's no listeneres interested in an activity or no listeners are registered, then `ActivitySource.StartActivity` will return `null`, hence `Activity.Current` will be `null` too.

So when there's no activity I don't know if just creating a new (random) Id is correct.
Maybe in that case it should just be empty?

Reply:
@jeremydmiller 
So should we set the current activity if it is null? or not start an activity at all?